Cannot compile query containing if without else

jq runs a query containing an if without an else without any issues:

```bash
$ jq --version
jq-1.7
$ echo '{"a":"a","b":"b"}' | jq 'if false then .a end'
{
  "a": "a",
  "b": "b"
}
```

The documentation of [if-then-else-end](https://jqlang.org/manual/#if-then-else-end) states:
> `if A then B end` is the same as `if A then B else . end`. That is, the else branch is optional, and if absent is the same as `.`. This also applies to `elif` with absent ending `else` branch.

### Current behaviour

When I try to compile the same query in jackson-jq, I get an exception:

```java
JsonQuery.compile("if false then .a end", Versions.JQ_1_7);
```
```
net.thisptr.jackson.jq.exception.JsonQueryException: Cannot compile query: if false then .a  end

	at net.thisptr.jackson.jq.internal.javacc.ExpressionParser.compile(ExpressionParser.java:34)
	at net.thisptr.jackson.jq.JsonQuery.compile(JsonQuery.java:21)
	at ch.livit.grem.eac.linter.LinterTest.foo(LinterTest.java:37)
Caused by: net.thisptr.jackson.jq.internal.javacc.ParseException: Encountered " "end" "end "" at line 1, column 18.
Was expecting one of:
    "and" ...
    "or" ...
    "//" ...
    "==" ...
    "!=" ...
    ">" ...
    ">=" ...
    "<" ...
    "<=" ...
    "+" ...
    "-" ...
    "*" ...
    "/" ...
    "%" ...
    "=" ...
    "|=" ...
    "+=" ...
    "-=" ...
    "*=" ...
    "/=" ...
    "%=" ...
    "//=" ...
    "." ...
    "[" ...
    "?" ...
    "|" ...
    "," ...
    "elif" ...
    "else" ...
    "as" ...
    
	at net.thisptr.jackson.jq.internal.javacc.ExpressionParser.generateParseException(ExpressionParser.java:2655)
	at net.thisptr.jackson.jq.internal.javacc.ExpressionParser.jj_consume_token(ExpressionParser.java:2515)
	at net.thisptr.jackson.jq.internal.javacc.ExpressionParser.ConditionalExpression(ExpressionParser.java:1063)
	at net.thisptr.jackson.jq.internal.javacc.ExpressionParser.PrimaryExpression(ExpressionParser.java:908)
	at net.thisptr.jackson.jq.internal.javacc.ExpressionParser.BinaryOperatorExpression(ExpressionParser.java:540)
	at net.thisptr.jackson.jq.internal.javacc.ExpressionParser.TupleExpression(ExpressionParser.java:513)
	at net.thisptr.jackson.jq.internal.javacc.ExpressionParser.PipeComponent(ExpressionParser.java:435)
	at net.thisptr.jackson.jq.internal.javacc.ExpressionParser.Expression(ExpressionParser.java:341)
	at net.thisptr.jackson.jq.internal.javacc.ExpressionParser.Start(ExpressionParser.java:331)
	at net.thisptr.jackson.jq.internal.javacc.ExpressionParser.compile(ExpressionParser.java:30)
	... 2 more
```

### Expected behaviour

jackson-jq should be able to compile and execute queries containing an `if` / `elif` without an `else` branch.

--- a/jackson-jq/src/main/javacc/json-query.jj
+++ b/jackson-jq/src/main/javacc/json-query.jj
@@ -775,7 +775,7 @@ Expression ConditionalExpression():
 	Expression tmpPred;
 	Expression tmpExpr;
 	final List<Pair<Expression, Expression>> switches = new ArrayList<Pair<Expression, Expression>>();
-	final Expression otherwise;
+	Expression otherwise = new ThisObject();
 }
 {
 	<KEYWORD_IF>
@@ -790,8 +790,10 @@ Expression ConditionalExpression():
 		tmpExpr = Expression()
 		{ switches.add(Pair.of(tmpPred, tmpExpr)); }
 	)*
-	<KEYWORD_ELSE>
-	otherwise = Expression()
+	(
+    	<KEYWORD_ELSE>
+    	otherwise = Expression()
+    )?
 	<KEYWORD_END>
 	{
 		return new Conditional(switches, otherwise);

--- a/jackson-jq/src/test/resources/tests/constructs/if.yaml
+++ b/jackson-jq/src/test/resources/tests/constructs/if.yaml
@@ -1,0 +1,14 @@
+- q: 'if true then 1 else 2 end'
+  out:
+  - 1
+- q: 'if false then 1 end'
+  in: 2
+  out:
+  - 2
+- q: 'if false then 1 elif false then 2 else 3 end'
+  out:
+  - 3
+- q: 'if false then 1 elif false then 2 end'
+  in: 3
+  out:
+  - 3

